test.py

```diff
@@ -1305,6 +1324,7 @@ def parse_cmd_line() -> argparse.Namespace:
         help="""Path to temporary test data and log files. The data is
         further segregated per build mode. Default: ./testlog.""",
     )
+    parser.add_argument("--gather_metrics", action=argparse.BooleanOptionalAction, default=False)
```

Comment:
The main reason is to not break the flow if somebody tries to execute the test.py outside the toolchain. Additionally, I doubt that is beneficial in the current state to switch on since it just metric gathering, and they are useless if nobody analyzing them.

Reply:
Why would it break the flow?

I like to keep CI using defaults, so developers aren't surprised when something works on their machine but doesn't work on CI.

We can add ~/.config/scylladb/scylladb-test-py.conf so people can adjust their configuration to taste.

Reply:
The idea was to switch it on upon need. Implemented logic will not work outside the toolchain. So with default set to true and using just test.py will not work because we have no power to manipulate the cgroups in the root. Even if I rewrite the logic to start from the cgroup where the process is, there still can be an issue that this cgroup will be owned by root, and we will not be able to create another cgroup or put the PID to the desired one.

Reply:
> The idea was to switch it on upon need. Implemented logic will not work outside the toolchain. 

Why not? Everything else does. Please don't write code so fragile it only works in one scenario.

> So with default set to true and using just test.py will not work because we have no power to manipulate the cgroups in the root. Even if I rewrite the logic to start from the cgroup where the process is, there still can be an issue that this cgroup will be owned by root, and we will not be able to create another cgroup or put the PID to the desired one.

I don't understand why. Podman is able to create cgroups without being root. So should we.

Reply:
> Why not? Everything else does. Please don't write code so fragile it only works in one scenario.

It's not fragile. It works in the podman and in the docker. Generally speaking, in the environment where we have control. Outside the container there can be any configuration, and it will be hardly possible to cover all of them.

> I don't understand why. Podman is able to create cgroups without being root. So should we.

We can create the cgroups as well without root in the cgroups where the user is an owner. We can assume that the process where tetst.py is started located in the cgroup with correct permission, but this is not always true. In that case, we need to find the cgroup with correct permissions. After that, the flow will be actually the same. But finding the correct cgroup can be tricky a little bit.

Reply:
Does this not work?

```console
$ cat /proc/$$/cgroup 
0::/user.slice/user-1000.slice/user@1000.service/app.slice/app-org.gnome.Terminal.slice/vte-spawn-aff2c052-ef9b-4023-afef-3bd36a81fa19.scope
```

Reply:
It works, but for example, my foot terminal instance will be in the cgroup where the root is owner. So in most cases it will be easy to go get cgroup through the /proc/self/cgroup, but still, we cannot rely fully on it.